src/util/MemorySize/MemorySize.h

```diff
@@ -327,23 +328,35 @@ constexpr double MemorySize::getTerabytes() const {
 
 // _____________________________________________________________________________
 constexpr MemorySize MemorySize::operator+(const MemorySize& m) const {
+  // Check for overflow.
+  if (memoryInBytes_ > size_t_max - m.memoryInBytes_) {
+    throw std::overflow_error(
+        "Overflow error: Addition of the two given 'MemorySize' not possible. "
+        "Would result in size_t overflow.");
```

Comment:
```suggestion
        "Overflow error: Addition of the two given 'MemorySize's is not possible. "
        "Would result in size_t overflow.");
```
And while you are at it you can integrate the two values in the exception message (same for the other operations).

Reply:
Done, see [d2a0cd2](https://github.com/ad-freiburg/qlever/pull/1083/commits/d2a0cd2552f5d9bd46a096674752e1822f43fcb6).